app/models/manageiq/providers/cloud_manager/template.rb

```diff
@@ -1,5 +1,10 @@
 class ManageIQ::Providers::CloudManager::Template < ::MiqTemplate
   default_value_for :cloud, true
+  virtual_column :snapshot_or_image, :type => :string
+
+  def snapshot_or_image
+    genealogy_parent.nil? ? _("Image") : _("Snapshot")
+  end
```

Comment:
This looks like UI decorator code.  I think the model can add boolean predicates -- `snapshot?` and `image?` methods.  These can then be consumed by UI code.